File: atividades/ativ-verificacao-java/Stack.java
package stack;

/**
 * Created by Mateus Antonio on 06/04/2016.
 */

public class Stack {
    int topo = -1;
    static final int MAXSIZE = 50;
    char[] pilha = new char[MAXSIZE];

    public void adicionarPilha(char e) {
        topo++;
        pilha[topo] = e;
    }

    public char topoPilha(){
        return pilha[topo];
    }

    public char removerPilha() {
        char aux = pilha[topo];
        topo--;
        return aux;
    }

    public boolean estaVazia(){
        return (topo==-1);
    }
}

File: atividades/ativ-verificacao-c/Stack.cpp
#include "Stack.h"

Stack::Stack()
{
    //ctor
}

void Stack::adicionarPilha(char e){
    topo++;
    pilha[topo] = e;
}

char Stack::removerPilha(){
    char aux = pilha[topo];
    topo--;
    return aux;
}

char Stack::topoPilha(){
    return pilha[topo];
}

bool Stack::estaVazia(){
    return (topo==-1);
}
